sound/soc/sof/ipc3.c

```diff
@@ -1023,18 +1038,12 @@ static void sof_ipc3_rx_msg(struct snd_sof_dev *sdev)
 	err = snd_sof_ipc_msg_data(sdev, NULL, msg_buf, hdr.size);
 	if (err < 0) {
 		dev_err(sdev->dev, "%s: Failed to read message: %d\n", __func__, err);
-	} else {
-		/* Call local handler for the message */
-		if (rx_callback)
-			rx_callback(sdev, msg_buf);
-
-		/* Notify registered clients */
-		sof_client_ipc_rx_dispatcher(sdev, msg_buf);
+		kfree(msg_buf);
+		return;
 	}
+	sof_ipc3_do_rx_work(sdev, &hdr, msg_buf);
```

Comment:
I don't understand why we need all this.

The client injector should do what you want already. What is missing in its existing implementation?

Can you clarify the purpose of this patch?

Reply:
Also we have a requirement on our side to fuzz the kernel to harden it against malicious firmware, which is why we are adding this.

Reply:
oh so you use the client injector to send stuff to the firmware but override the replies from the firmware with your side path?

Reply:
@plbossart I think there's no TX here at all. I think they're trying to fuzz only the rx from the DSP part without any IPCs sent to the DSP

Reply:
@ranj063 is correct, this is fuzzing a DSP originated message, replies are not in scope currently.

Reply:
No sure I understand the scope of the testing then. If there is nothing really consuming the information, what does this accomplish? Is this just to test the DSP notification and DSP-initiated alerts? Isn't that a very very small part of what the IPC protocol is about.

I would have sent a fake message and inserted a fake reply to cause all sorts of trouble.

Reply:
@plbossart I think the scope of fuzzing can include both fake messages injection in both directions but done independently. With this PR, the injector client can handle both.

Reply:
also we rarely use replies